pkg/controller/migmigration/task.go

```diff
@@ -945,13 +945,11 @@ func (t *Task) initPipeline() {
 	}
 }
 
-func (t *Task) updatePipeline(prevStep string) {
-	oldStep := t.Owner.Status.FindStep(prevStep)
+func (t *Task) updatePipeline() {
 	currentStep := t.Owner.Status.FindStep(t.Step)
-	if oldStep != nil && oldStep != currentStep {
-		oldStep.MarkCompleted()
-		if t.failed() {
-			oldStep.Failed = true
+	for _, step := range t.Owner.Status.Pipeline {
```

Comment:
@djwhatle This makes sure all previous steps are marked complete when we move to the next step.

Reply:
+1